Ginger/GingerCore/ALM/Qtest/ImportFromQtest.cs

```diff
@@ -0,0 +1,1128 @@
+                    if (repoStepActivity != null)
+                    {
+                        //check if it is part of the Activities Group
+                        ActivityIdentifiers groupStepActivityIdent = (ActivityIdentifiers)tcActivsGroup.ActivitiesIdentifiers.Where(x => x.ActivityExternalID == step.StepID).FirstOrDefault();
+                        if (groupStepActivityIdent != null)
+                        {
+                            //already in Activities Group so get link to it
+                            stepActivity = (Activity)busFlow.Activities.Where(x => x.Guid == groupStepActivityIdent.ActivityGuid).FirstOrDefault();
+                        }
+                        else//not in ActivitiesGroup so get instance from repo
+                        {
+                            stepActivity = (Activity)repoStepActivity.CreateInstance();
+                            toAddStepActivity = true;
+                        }
+                        stepActivity.IsSharedRepositoryInstance = true;
+                    }
+                    else//Step not exist in Ginger repository so create new one
+                    {
+                        stepActivity = new Activity();
+                        stepActivity.ActivityName = tc.TestName + ">" + step.StepName;
+                        stepActivity.ExternalID = step.StepID;
+                        stepActivity.Description = StripHTML(step.Description);
+                        stepActivity.Expected = StripHTML(step.Expected);
+
+                        toAddStepActivity = true;
+                    }
+
+                    if (toAddStepActivity)
+                    {
+                        //not in group- need to add it
+                        busFlow.AddActivity(stepActivity, tcActivsGroup, startGroupActsIndxInBf++);                        
+                    }
+
+                    //pull TC-Step parameters and add them to the Activity level
+                    List<string> stepParamsList = new List<string>();
+                    GetStepParameters(StripHTML(step.Description), ref stepParamsList);
+                    GetStepParameters(StripHTML(step.Expected), ref stepParamsList);
+                    foreach (string param in stepParamsList)
+                    {
+                        //get the param value
+                        string paramSelectedValue = string.Empty;
+                        bool? isflowControlParam = null;
+                        QtestTestParameter tcParameter = tc.Parameters.Where(x => x.Name.ToUpper() == param.ToUpper()).FirstOrDefault();
+
+                        //get the param value
+                        if (tcParameter != null && tcParameter.Value != null && tcParameter.Value != string.Empty)
+                            paramSelectedValue = tcParameter.Value;
+                        else
+                        {
+                            isflowControlParam = null;//empty value
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)